core/extensions/src/script_hash/mod.rs

```diff
@@ -27,16 +27,50 @@ impl<S: Store, BS: Store> Extension for ScriptHashExtension<S, BS> {
         let block_hash = block.hash();
         let mut batch = self.store.batch()?;
 
-        for tx in block.transactions().iter() {
+        for (tx_index, tx) in block.transactions().iter().enumerate() {
             let tx_hash: [u8; 32] = tx.hash().unpack();
             batch.put_kv(
                 Key::TxHash(tx_hash),
                 Value::BlockNumAndHash(block_num, &block_hash),
             )?;
 
-            for output in tx.outputs().into_iter() {
+            if tx_index > 0 {
+                for (io_index, input) in tx.inputs().into_iter().enumerate() {
+                    let out_point = input.previous_output();
+                    let cell_index: u32 = out_point.index().unpack();
+                    let cell = if block.tx_hashes().contains(&out_point.tx_hash()) {
+                        tx.outputs().get(cell_index as usize).unwrap()
+                    } else {
+                        let detailed_cell = self.get_live_cell_by_out_point(&out_point)?;
```

Comment:
The `extension.append` happens after `indexer.append`, so it's likely would not find live_cell here. ref: https://github.com/nervosnetwork/mercury/blob/main/core/service/src/lib.rs#L173-L178